### PR TITLE
Creating unit tests and running them as part of PRs

### PR DESCRIPTION
**Pull request by edrickrenan**
## Description

The idea behind this PR is to add real unit tests, knowing that existing ones depend on a person's original config.json.

With this change, I add mocks that return exactly what the Binance API returns. Also, I add the workflow to execute the tests every time a branch is merged into master or a PR is created to master.

Note: I haven't covered all the possible scenarios there, it's only an initiative, and we can start adding tests as we make changes.

## Type of change

Please make your selection.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change requires a new dependency
- [x] Code Maintainability / comments
- [ ] Code Refactoring / future-proofing

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added pytest unit tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by edrickrenan** [comment]
https://github.com/whittlem/pycryptobot/runs/2575306018?check_suite_focus=true

File: tests/unit_tests/test_Model_Binance.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+            api_url = config['binance']['api_url']
+            AuthAPI(api_key, api_secret, api_url)
+
+    exchange = AuthAPI(api_key, api_secret, api_url)
+    assert type(exchange) is AuthAPI
+
+    mocker.patch("models.Binance.Client.get_trade_fee", return_value=client_response)
+    fee = exchange.getTakerFee(MOCK_MARKET)
+    assert type(fee) is float
+    assert fee == 0.001
```
Comment (edrickrenan) [inline tests/unit_tests/test_Model_Binance.py:184]: Good thing is that now we can assert the exact value we expect. In the original tests, it is `fee > 0`, so if the `exchange.getTakerFee` returns anything different than `0.001`, we know that the change is breaking something.

Comment (whittlem) [reply tests/unit_tests/test_Model_Binance.py:184]: Sounds great.

**Review by whittlem — APPROVED** [review]

**Comment by whittlem** [comment]
This is great, thanks.

**Pull request merged**